### PR TITLE
InputCommon: Fix offbrand GCAdapter

### PR DESCRIPTION
**Pull request by german77**
Allows offbrand GCAdapters to work, port from https://github.com/dolphin-emu/dolphin/pull/8933

### CODE REVIEW

File: src/input_common/gcadapter/gc_adapter.cpp [hunk 1]
```diff
@@ -218,6 +217,12 @@ bool Adapter::CheckDeviceAccess(libusb_device* device) {
         }
     }
 
+    int control_transfer_error =
+        libusb_control_transfer(usb_adapter_handle, 0x21, 11, 0x0001, 0, nullptr, 0, 1000);
```
Comment (ameerj) [inline src/input_common/gcadapter/gc_adapter.cpp:221]: Just out of curiosity, what are these numbers?

File: src/input_common/gcadapter/gc_adapter.cpp [hunk 2]
```diff
@@ -218,6 +217,12 @@ bool Adapter::CheckDeviceAccess(libusb_device* device) {
         }
     }
 
+    int control_transfer_error =
```
Comment (lioncash) [inline src/input_common/gcadapter/gc_adapter.cpp:220]:
```suggestion
    const int control_transfer_error =
```

File: src/input_common/gcadapter/gc_adapter.cpp (hunk 1 above)
Comment (german77) [reply src/input_common/gcadapter/gc_adapter.cpp:221]: Magic numbers. I haven't done any research just added the fix from dolphin.

Comment (ameerj) [reply src/input_common/gcadapter/gc_adapter.cpp:221]: Can you please add a comment just to clarify the purpose of this line for anyone looking at the code later on.

**Pull request closed without merging**